src/espIdf/setTarget/index.ts

```diff
@@ -169,23 +188,33 @@ export async function setIdfTarget(
             configurationTarget,
             workspaceFolder.uri
           );
-          // Store USB location if available
+          // Store USB location if available (will be used as fallback if serial is not found)
           if (selectedTarget.boardInfo.location) {
-            const customExtraVars = readParameter(
-              "idf.customExtraVars",
-              workspaceFolder
-            ) as { [key: string]: string };
             const location = selectedTarget.boardInfo.location.replace(
               "usb://",
               ""
             );
             customExtraVars["OPENOCD_USB_ADAPTER_LOCATION"] = location;
-            await writeParameter(
-              "idf.customExtraVars",
-              customExtraVars,
-              configurationTarget,
+          }
+          
+          // Update serial port if board is connected
+          // The serial port should match the connected board
+          try {
+            const detectedPort = await SerialPort.detectDefaultPort(
               workspaceFolder.uri
             );
+            if (detectedPort) {
```

Comment:
I can see that we are using detectDefaultPort here to get the first chip that matches selected IDF_TARGET.

This doesn't mean that the first chip detected is the same as the one from `OPENOCD_USB_ADAPTER_LOCATION`. This could be true for only one of same chip connected, but **what happens when we have multiple esp32 connected at the same time?**

Can the serial port (for example `/dev/cu*` be extracted from `selectedTarget.boardInfo` directly ? If yes, we can just update the idf.port setting directly.

if we can't get the serial port from `selectedTarget.boardInfo`, we need to find a way to map `selectedTarget.boardInfo.location` to one of serial ports given by `SerialPort.shared().list()` since I think `SerialPort.detectDefaultPort` will not necessarily return the right port always.

Reply:
I've removed the functionality of correlating the serial port with the usb adapter location. Changing the target, won't do anything to the serial port settings anymore. (Back to how it was before this PR)